### PR TITLE
UICIRC-909 - Implement General information and Template content accordions in edit view of Staff slip

### PR DESCRIPTION
**Pull request by Terala-Priyanka**
## Purpose
Implement General information and Template content accordions in edit view of Staff slip

## Refs
https://issues.folio.org/browse/UICIRC-910
https://issues.folio.org/browse/UICIRC-909

## Screenshots
xvR3NbUt4S

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by Terala-Priyanka** [comment]
@folio-org/fe-tl-reviewers Please take some time to review this PR

**Comment by Terala-Priyanka** [comment]
@folio-org/fe-tl-reviewers Could you please take some time to review this PR?

File: src/settings/StaffSlips/components/EditSections/StaffSlipTemplateContentSection/StaffSlipTemplateContentSection.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  injectIntl,
+} from 'react-intl';
+import { Field } from 'react-final-form';
+
+import {
+  Col,
+  Row,
+} from '@folio/stripes/components';
+import { TemplateEditor } from '@folio/stripes-template-editor';
+import TokensList from '../../../TokensList';
```
Comment (mariia-aloshyna) [inline src/settings/StaffSlips/components/EditSections/StaffSlipTemplateContentSection/StaffSlipTemplateContentSection.js:13]: It looks like you separated stripes and local imports by empty line in other files, should this be also separated for consistency?

**Review by mariia-aloshyna — APPROVED** [review]

File: src/settings/StaffSlips/StaffSlipForm.test.js [hunk 2]
```diff
@@ -9,63 +9,100 @@ import '../../../test/jest/__mock__';
...
+  AccordionSet,
+  ExpandAllButton,
 } from '@folio/stripes/components';
 
 import StaffSlipForm from './StaffSlipForm';
-import TokensList from './TokensList';
-import getTokens from './tokens';
+import {
+  StaffSlipAboutSection,
+  StaffSlipTemplateContentSection
```
Comment (artem-blazhko) [inline src/settings/StaffSlips/StaffSlipForm.test.js:23]:
```suggestion
  StaffSlipTemplateContentSection,
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -21,6 +21,8 @@
 * UI tests replacement with RTL/Jest for component `RulesEditor`. Refs UICIRC-827.
 * UI tests replacement with RTL/Jest for `Rules-hint`. Refs UICIRC-831.
 * Add metadata information to view of Staff Slips (Settings > Circulation > Staff Slips). Refs UICIRC-855.
+* Implement General information accordion. Refs UICIRC-909
```
Comment (Dmitriy-Litvinenko) [inline CHANGELOG.md:24]: Could you please update CHANGELOG.md (example below)?
```
## IN PROGRESS

* Add metadata information to view of Staff Slips (Settings > Circulation > Staff Slips). Refs UICIRC-855.
* Implement General information accordion. Refs UICIRC-909
* Implement Template content accordion. Refs UICIRC-910
 
## [8.0.0](https://github.com/folio-org/ui-circulation/tree/v8.0.0) (2023-02-22)
[Full Changelog](https://github.com/folio-org/ui-circulation/compare/v7.2.1...v8.0.0)

* Make consistent view of delete functionality of "Fixed due date schedules". Refs UICIRC-874.
...
* Add metadata information to view of Staff Slips (Settings > Circulation > Staff Slips). Refs UICIRC-855. // remove
```

File: src/settings/StaffSlips/components/EditSections/StaffSlipAboutSection/StaffSlipAboutSection.test.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+  const getItemByTestId = (id) => within(screen.getByTestId(id));
+
+  const labelIds = {
+    staffSlipsName: 'ui-circulation.settings.staffSlips.name',
+    staffSlipsDescription: 'ui-circulation.settings.staffSlips.description',
+  };
+
+  const testIds = {
+    staffSlipAboutSectionTestId: 'staffSlipAboutSectionTestId',
```
Comment (artem-blazhko) [inline src/settings/StaffSlips/components/EditSections/StaffSlipAboutSection/StaffSlipAboutSection.test.js:26]: Do you use `'staffSlipAboutSectionTestId'` somewhere?

File: src/settings/StaffSlips/components/EditSections/StaffSlipTemplateContentSection/StaffSlipTemplateContentSection.js (hunk 1 above)
Comment (artem-blazhko) [reply src/settings/StaffSlips/components/EditSections/StaffSlipTemplateContentSection/StaffSlipTemplateContentSection.js:13]: @Terala-Priyanka could you consider the comment above?

File: src/settings/StaffSlips/components/EditSections/StaffSlipTemplateContentSection/StaffSlipTemplateContentSection.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  );
+};
+
+StaffSlipTemplateContentSection.propTypes = {
+  intl: PropTypes.object.isRequired,
+  staffSlip: PropTypes.shape({
+    name: PropTypes.string,
+    description: PropTypes.string,
+    template: PropTypes.string,
+  }).isRequired,
```
[the remark this answers is not in the document]
Comment (Terala-Priyanka) [reply src/settings/StaffSlips/components/EditSections/StaffSlipTemplateContentSection/StaffSlipTemplateContentSection.js:46]: looks like I have removed it while refining the code.. Thank you for the catch :+1:

File: CHANGELOG.md (hunk 3 above)
Comment (Terala-Priyanka) [reply CHANGELOG.md]: Thanks for catching it @Dmitriy-Litvinenko 
Corrected it

**Review by artem-blazhko — APPROVED** [review]

**Pull request merged**